lib/logstash/filters/mutate.rb

```diff
@@ -338,85 +334,91 @@ def gsub_dynamic_fields(event, original, needle, replacement)
     end
   end
 
-  private
   def uppercase(event)
     @uppercase.each do |field|
-      if event[field].is_a?(Array)
-        event[field].each { |v| v.upcase! }
-      elsif event[field].is_a?(String)
-        event[field].upcase!
-      else
-        @logger.debug("Can't uppercase something that isn't a string",
-                      :field => field, :value => event[field])
-      end
+      original = event[field]
+      event[field] = case original
+        when Array
+          original.map(&:upcase!)
```

Comment:
`upcase!` modifies the string itself, so either we don't need to assign back to `event[field]` or we could avoid using the `!` form of upcase?

```
>> a = "hello"
>> a.upcase!
>> a
"HELLO"
```